psl-core/src/main/java/org/linqs/psl/application/learning/weight/search/bayesian/GaussianProcessPrior.java

```diff
@@ -49,15 +59,15 @@ public GaussianProcessPrior(List<Rule> rules, Database rvDB, Database observedDB
         exploration = Options.WLA_GPP_EXPLORATION.getFloat();
         randomConfigsOnly = Options.WLA_GPP_RANDOM_CONFIGS_ONLY.getBoolean();
         earlyStopping = Options.WLA_GPP_EARLY_STOPPING.getBoolean();
+        useProvidedWeight = Options.WLA_GPP_USE_PROVIDED_WEIGHT.getBoolean();
 
-        initialWeightValue = Options.WLA_GPP_INITIAL_WEIGHT_VALUE.getFloat();
+        initialWeightValue = evaluator.isHigherRepBetter()? Options.WLA_GPP_INITIAL_WEIGHT_VALUE.getFloat()
```

Comment:
I changed the names of these variables to reflect what they are truly for. The initialWeightValue was not the actual wait but instead the expected metric value at that location, initially, this is shared for all configurations. 

I do the flip here because the function value returned by the evaluator method getNormalizedRepMetric() which updates configuration metric values always returns values so that higher is better, and this characteristic is assumed for the comparisons in the GPP code. I could flip the value returned by getNormalizedRepMetric() again, and then perform checks for isHigherBetter before comparisons or stick with this pattern.